Dockerfile

```diff
@@ -1,10 +1,38 @@
-FROM ruby:2.5.1
-RUN apt-get update -qq && apt-get install -y ruby-dev
+# Install gems
+FROM alpine:3.8 as build
+
+RUN \
+apk --no-cache add build-base ruby-dev ruby-bundler ruby-json ruby-bigdecimal git openssl-dev && \
+echo 'gem: --no-document' > /etc/gemrc && \
+bundle config --global silence_root_warning 1
 
 RUN mkdir /bolt-server
-WORKDIR /bolt-server
+# Gemfile requires gemspec which requires bolt/version which requires bolt
 ADD . /bolt-server
-RUN bundle install
+WORKDIR /bolt-server
+RUN bundle install --no-cache --path vendor/bundle
+
+# Final image
+FROM alpine:3.8
+ARG bolt_version=no-version
+LABEL org.label-schema.maintainer="Puppet Bolt Team <team-direct-change-bolt@puppet.com>" \
+      org.label-schema.vendor="Puppet" \
+      org.label-schema.url="https://github.com/puppetlabs/bolt" \
+      org.label-schema.name="PE Bolt Server" \
+      org.label-schema.license="Apache-2.0" \
+      org.label-schema.version=${bolt_version} \
+      org.label-schema.vcs-url="https://github.com/puppetlabs/bolt" \
+      # Same with these
```

Comment:
What does `Same with these` mean?

Reply:
Oh, I was also not sure how to programmatically set them. @adreyer  Are there similar variables that get set that I can put here?